client/analytics/report/products/index.js

```diff
@@ -62,3 +126,34 @@ ProductsReport.propTypes = {
 	path: PropTypes.string.isRequired,
 	query: PropTypes.object.isRequired,
 };
+
+export default compose(
+	withSelect( ( select, props ) => {
+		const { query } = props;
+		const { getProducts, isGetProductsRequesting, getProductsError } = select( 'wc-api' );
+		const isSingleProductView = query.products && 1 === query.products.split( ',' ).length;
+		if ( isSingleProductView ) {
+			const includeArgs = { include: query.products };
+			const products = getProducts( includeArgs );
```

Comment:
We make this exact same request to populate tags in the Search component. Just noting that there is an opportunity here (in another PR).

Reply:
I left a `TODO` for this in https://github.com/woocommerce/wc-admin/pull/1391/commits/275356e15585f88dc9f21d0a676dc6e111633c0d.